File: src/main/java/ru/twilson/tasktracker/entity/Task.java
package ru.twilson.tasktracker.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Entity
@Builder(toBuilder = true)
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class Task {

    @Id
    @GeneratedValue
    private long taskId;

    @Column(unique = true, nullable = false)
    private String taskGlobalId;
    private String title;
    private String description;
    private String priority;
    private String dueDate;
    private String createdAt;
    private String updatedAt;
    private String completedAt;
    private String status;
    private boolean isDeleted = false;

    @ManyToOne(cascade = CascadeType.ALL)
    private Consumer creator;

    @ManyToOne(cascade = CascadeType.ALL)
    private Consumer executor;


    public Task copy() {
        return Task.builder()
                .taskId(taskId)
                .taskGlobalId(taskGlobalId)
                .title(title)
                .description(description)
                .priority(priority)
                .dueDate(dueDate)
                .createdAt(createdAt)
                .updatedAt(updatedAt)
                .completedAt(completedAt)
                .status(status)
                .creator(creator)
                .executor(executor)
                .build();
    }
}

File: ui/TaskTracker/Models/Task.cs
using System;

namespace TaskTracker.Models;

public class Task
{
    public Guid Id { get; set; }

    public string Title { get; set; }

    public string Description { get; set; }

    public string? Status { get; set; }

    public string Priority { get; set; }

    public DateTime DueDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTime? CompletedAt { get; set; }

    public Guid UserId { get; set; }

    public Task(
        Guid id,
        string title,
        string description,
        string? status,
        string priority,
        DateTime dueDate,
        DateTime? createdAt,
        DateTime? updatedAt,
        DateTime? completedAt,
        Guid userId)
    {
        Id = id;
        Title = title;
        Description = description;
        Status = status;
        Priority = priority;
        DueDate = dueDate;
        CreatedAt = createdAt;
        UpdatedAt = updatedAt;
        CompletedAt = completedAt;
        UserId = userId;
    }
}